src/TwigTemplate.php

```diff
@@ -23,4 +26,20 @@ public function render( array $arguments ): string {
 		return $this->twig->render( $this->templatePath, $arguments + $this->context );
 	}
 
+	/**
+	 * When using this method, arguments can be rendered as templates by
+	 * using {% include 'argumentName' %} instead of {$ argumentName $}.
+	 *
+	 * @param array $arguments
+	 * @return string
+	 */
+	public function renderArgumentsAsTemplates( array $arguments ): string {
+		$loader = new Twig_Loader_Chain( [
+			new Twig_Loader_Array( $arguments ),
+			$this->twig->getLoader()
+		] );
+		$this->twig->setLoader( $loader );
```

Comment:
Like this a dependency that was constructor injected gets modified on calling this method. That's unexpected state change. Is there no way to avoid this?